Tests/Basic/PathTests.swift

```diff
@@ -0,0 +1,177 @@
+        XCTAssertEqual(AbsolutePath("/a").basename, "a")
+        XCTAssertEqual(AbsolutePath("/./a").basename, "a")
+        XCTAssertEqual(AbsolutePath("/../..").basename, "/")
+        XCTAssertEqual(RelativePath("../..").basename, "..")
+        XCTAssertEqual(RelativePath("../a").basename, "a")
+        XCTAssertEqual(RelativePath("../a/..").basename, "..")
+        XCTAssertEqual(RelativePath("a/..").basename, ".")
+        XCTAssertEqual(RelativePath("./..").basename, "..")
+        XCTAssertEqual(RelativePath("a/../////../////./////").basename, "..")
+        XCTAssertEqual(RelativePath("abc").basename, "abc")
+        XCTAssertEqual(RelativePath("").basename, ".")
+        XCTAssertEqual(RelativePath(".").basename, ".")
+    }
+    
+    func testSuffixExtraction() {
+        XCTAssertEqual(RelativePath("a").suffix, nil)
+        XCTAssertEqual(RelativePath("a.").suffix, nil)
+        XCTAssertEqual(RelativePath(".a").suffix, nil)
+        XCTAssertEqual(RelativePath("").suffix, nil)
+        XCTAssertEqual(RelativePath(".").suffix, nil)
+        XCTAssertEqual(RelativePath("..").suffix, nil)
+        XCTAssertEqual(RelativePath("a.foo").suffix, ".foo")
+        XCTAssertEqual(RelativePath(".a.foo").suffix, ".foo")
+        XCTAssertEqual(RelativePath(".a.foo.bar").suffix, ".bar")
+        XCTAssertEqual(RelativePath("a.foo.bar").suffix, ".bar")
+        XCTAssertEqual(RelativePath(".a.foo.bar.baz").suffix, ".baz")
+    }
+    
+    func testParentDirectory() {
+        XCTAssertEqual(AbsolutePath("/").parentDirectory, AbsolutePath("/"))
+        XCTAssertEqual(AbsolutePath("/").parentDirectory.parentDirectory, AbsolutePath("/"))
+        XCTAssertEqual(AbsolutePath("/bar").parentDirectory, AbsolutePath("/"))
+        XCTAssertEqual(AbsolutePath("/bar/../foo/..//").parentDirectory.parentDirectory, AbsolutePath("/"))
+        XCTAssertEqual(AbsolutePath("/bar/../foo/..//yabba/a/b").parentDirectory.parentDirectory, AbsolutePath("/yabba"))
+    }
+    
+    func testConcatenation() {
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/"), RelativePath("")).asString, "/")
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/"), RelativePath(".")).asString, "/")
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/"), RelativePath("..")).asString, "/")
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/"), RelativePath("bar")).asString, "/bar")
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/bar"), RelativePath("../foo")).asString, "/foo")
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/bar"), RelativePath("../foo/..//")).asString, "/")
+        XCTAssertEqual(AbsolutePath(AbsolutePath("/bar/../foo/..//yabba/"), RelativePath("a/b")).asString, "/yabba/a/b")
+    }
+    
+    // FIXME: We also need tests for join() operations.
+    
+    // FIXME: We also need tests for dirname, basename, suffix, etc.
+    
+    // FIXME: We also need test for stat() operations.
+        
+    static var allTests = [
```

Comment:
this needs to be added to Basic/XCTestManifests.swift to be able to run on Linux